test/adapter/json_api/relationship_test.rb

```diff
@@ -150,9 +150,23 @@ def test_relationship_with_everything
 
         private
 
-        def test_relationship(expected, params = {})
+        def test_relationship(expected, test_options = {})
           parent_serializer = AuthorSerializer.new(@author)
-          relationship = Relationship.new(parent_serializer, @serializer, nil, params)
+
+          serializable_resource_options = {} # adapter.instance_options
+
+          meta = test_options.delete(:meta)
+          options = test_options.delete(:options)
+          links = test_options.delete(:links)
+          association_serializer = @serializer
+          if association_serializer && association_serializer.object
+            association_name = association_serializer.json_key.to_sym
+            association = ::ActiveModel::Serializer::Association.new(association_name, association_serializer, options, links, meta)
+          else
+            association = ::ActiveModel::Serializer::Association.new(:association_name_not_used, association, options, links, meta)
+          end
+
+          relationship = Relationship.new(parent_serializer, serializable_resource_options, association)
           assert_equal(expected, relationship.as_json)
         end
```

Comment:
(copied from https://github.com/rails-api/active_model_serializers/pull/1884#discussion_r75056309 )

I ended up making the association manually here 'cause I couldn't figure out how else to get it.

Also tried:

```ruby
            include_directive = JSONAPI::IncludeDirective.new('*', allow_wildcard: true)
          associations = parent_serializer.associations(include_directive)
          association = associations.to_a.detect {|assoc| assoc.name == association_name }
          reflection = serializer._reflections[association_name]
          reflection.link(name, links) if links
          reflection.meta(meta) if meta
          reflection.include_data(options[:include_data]) if options
```

and something with `parent_serializer._reflections.select {|reflection| reflection.name == relationship_name }.build_association` but the way the test is written is kind of weird that our assertions depend on a magical `@author` model and `@serializer` of the desired association, but that doesn't very closely parallel how associations are build from the first serializer...